src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -161,6 +162,23 @@ public void setExpToDrop(int exp)
     }
 
     /**
+     * Called when a block is destroyed by an explosion.
+     * Canceling this event will prevent the block from being destroyed.
+     */
+    @Cancelable
+    public static class ExplodeEvent extends BlockEvent
+    {
+        @Nonnull private final Explosion explosion;
+
+        public ExplodeEvent(@Nonnull World world, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull Explosion explosion) {
+            super(world, pos, state);
+            this.explosion = explosion;
+        }
+
+        @Nonnull public Explosion getExplosion() { return explosion; }
```

Comment:
Formatting. Don't sqeeze it in one line, follow forge code style. Use this formatter: https://github.com/ForgeEssentials/ForgeEssentials/blob/develop/misc/eclipseformatter.xml